pkg/handlers/license.go

```diff
@@ -309,11 +310,63 @@ func (h *Handler) UploadNewLicense(w http.ResponseWriter, r *http.Request) {
 		JSON(w, 500, uploadLicenseResponse)
 		return
 	}
+
+	installedAppSet := make(map[string]bool)
 	if existingLicense != nil {
-		uploadLicenseResponse.Error = "License already exists"
-		uploadLicenseResponse.DeleteAppCommand = fmt.Sprintf("kubectl kots remove %s -n %s --force", existingLicense.Spec.AppSlug, util.PodNamespace)
-		JSON(w, 400, uploadLicenseResponse)
-		return
+		apps, err := store.GetStore().ListInstalledApps()
+		if err != nil {
+			logger.Error(errors.Wrap(err, "failed to list installed apps"))
+			uploadLicenseResponse.Error = err.Error()
+			JSON(w, 500, uploadLicenseResponse)
+		}
+
+		// build set of installed app slugs using this license
+		for _, app := range apps {
+			decode := scheme.Codecs.UniversalDeserializer().Decode
+			obj, _, err := decode([]byte(app.License), nil, nil)
+			if err != nil {
+				continue
+			}
+			license := obj.(*kotsv1beta1.License)
+
+			if license.Spec.LicenseID == verifiedLicense.Spec.LicenseID {
+				installedAppSet[app.Slug] = true
+			}
+		}
+
+		appRemoved := false
+		// get correct slug for app if it is not installed
+		titleForSlug := strings.Replace(verifiedLicense.Spec.AppSlug, ".", "-", 0)
+		slugProposal := slug.Make(titleForSlug)
+
+		i := 0
+		for {
+			if i > 0 {
+				slugProposal = fmt.Sprintf("%s-%d", titleForSlug, i)
+			}
+
+			// if slug is not installed attempt to remove
+			if !installedAppSet[slugProposal] {
+				if id, err := store.GetStore().GetAppIDFromSlug(slugProposal); err == nil && id != "" {
+					if err := store.GetStore().RemoveApp(id); err != nil {
+						logger.Error(errors.Wrap(err, "failed to remove app"))
+						uploadLicenseResponse.Error = err.Error()
```

Comment:
let's not return internal errors stack trace to the user
```suggestion
						uploadLicenseResponse.Error = "Failed to remove existing app record"
```